[quant][fx] EmbeddingBag quantization support

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46678 [quant][fx] EmbeddingBag quantization support**
* #46677 [quant][fx] Embedding quantization support

Summary:

Test Plan:
python test/test_quantization.py TestQuantzeFxOps.test_qembedding_bag_module

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24463306](https://our.internmc.facebook.com/intern/diff/D24463306)

supriyar committed:
[quant][fx] EmbeddingBag quantization support
Summary:

Test Plan:
python test/test_quantization.py TestQuantzeFxOps.test_qembedding_bag_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -398,6 +398,26 @@ def convert(self, quantizer, node, load_arg, debug=False, convert_custom_config_
             self.emb_node.target,
             (load_arg(quantized=False)(self.emb_node.args[0]),), {})
 
+@register_quant_pattern(torch.nn.EmbeddingBag)
```
looks like this can be merged with the nn.Embedding op

supriyar committed:
Update on "[quant][fx] EmbeddingBag quantization support"


Summary:

Test Plan:
python test/test_quantization.py TestQuantzeFxOps.test_qembedding_bag_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -389,7 +390,10 @@ def convert(self, quantizer, node, load_arg, debug=False, convert_custom_config_
         qconfig = quantizer.qconfig_map[node.name]
         assert not activation_is_statically_quantized(qconfig)
-        qemb = nnq.Embedding
+        if type(emb) == torch.nn.Embedding:
```
you can use: get_static_quant_module_class for now.. we can create a new map in the future.

jerryzh168 replied:
also please wait a bit until https://github.com/pytorch/pytorch/pull/46338 lands and add similar support here as well

supriyar committed:
Update on "[quant][fx] EmbeddingBag quantization support"


Summary:

Test Plan:
python test/test_quantization.py TestQuantzeFxOps.test_qembedding_bag_module

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24463306](https://our.internmc.facebook.com/intern/diff/D24463306)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] EmbeddingBag quantization support"


Summary:

Test Plan:
python test/test_quantization.py TestQuantzeFxOps.test_qembedding_bag_module

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24463306](https://our.internmc.facebook.com/intern/diff/D24463306)

[ghstack-poisoned]
